README.md

```diff
@@ -34,31 +34,53 @@
 
 ## IPFS Companion features
 
 ### Automatically detect and redirect IPFS resources
 
 #### Detect URLs with IPFS paths
 
-IPFS Companion detects and tests requests for IPFS-like paths ( such as `/ipfs/{cid}` or `/ipns/{peerid_or_host-with-dnslink}`) on any website. If a path is a [valid IPFS address](https://github.com/ipfs/is-ipfs), it is redirected to load from your local gateway. The gateway at `localhost` will also automatically switch to a subdomain to provide a unique origin for each website:
-> `https://ipfs.io/ipfs/QmbWqxBEKC3P8tqsKc98xmWNzrzDtRLMiMPL8wBuTGsMnR`
-> → `http://localhost:8080/ipfs/QmbWqxBEKC3P8tqsKc98xmWNzrzDtRLMiMPL8wBuTGsMnR`
-> → `http://bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi.ipfs.localhost:8080`
+IPFS Companion detects and tests requests for IPFS-like paths (such as `/ipfs/{cid}` or `/ipns/{peerid_or_host-with-dnslink}`) on any website. If a path is a [valid IPFS address](https://github.com/ipfs/is-ipfs), it is redirected to load from your local gateway. The gateway at `localhost` will also automatically switch to a [subdomain gateway](https://docs.ipfs.tech/how-to/address-ipfs-on-web/#subdomain-gateway) to provide a unique origin for each website:
+
+> [!NOTE]
+> **Path gateway redirect flow:**
+> 
+> - **Step 1:** Public gateway URL detected
+>   - `https://ipfs.io/ipfs/QmbWqxBEKC3P8tqsKc98xmWNzrzDtRLMiMPL8wBuTGsMnR`
+> - **Step 2:** Redirect to local gateway
+>   - `http://localhost:8080/ipfs/QmbWqxBEKC3P8tqsKc98xmWNzrzDtRLMiMPL8wBuTGsMnR`
+> - **Step 3:** Automatic upgrade to subdomain with origin isolation
+>   - `http://bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi.ipfs.localhost:8080`
+
+> [!NOTE]
+> **Subdomain gateway redirect flow:**
+> 
+> - **Step 1:** Subdomain gateway URL detected
+>   - `https://bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi.ipfs.dweb.link`
+> - **Step 2:** Redirect to local subdomain with origin isolation
+>   - `http://bafybeigdyrzt5sfp7udm7hu76uh7y26nf3efuylqabf3oclgtqy55fbzdi.ipfs.localhost:8080`
 
 #### Detect DNSLink-enabled URLs
 
-IPFS Companion [detects DNSLink info](http://docs.ipfs.io/how-to/dnslink-companion/) in the DNS records of websites. If a site uses DNSLink (a few examples are https://docs.ipfs.io, https://ipld.io, and http://tr.wikipedia-on-ipfs.org), IPFS Companion redirects the HTTP request to your local gateway:
+IPFS Companion [detects DNSLink info](https://docs.ipfs.tech/how-to/dnslink-companion/) in the DNS records of websites. If a site uses DNSLink (a few examples are https://docs.ipfs.tech, https://ipld.io, and http://tr.wikipedia-on-ipfs.org), IPFS Companion redirects the HTTP request to your local gateway:
```

Comment:
and vitalik.eth? it could be good to call out some other major players instead of only links we own.

Reply:
Let's keep as-is for this PR:
- `.eth` introduces world of pain and UX papercuts:
  - non-ICANN, `http://vitalik.eth` does not work  as expected  (no redirect to local gateway) because 
    - ICANN versions  like `_dnslink.vitalik.eth.limo` and `_dnslink.vitalik.eth.link` do not return any TXT records, which means companion cant detect they are on IPFS and cant redirect to local gateway without extra ENS-specific and domain-specific code